lib/fxpay/products.js

```diff
@@ -8,39 +8,53 @@
   var utils = fxpay.getattr('utils');
   var API = fxpay.getattr('api').API;
 
-  exports.all = function products_all(onResult) {
-    var allProducts = [];
-
-    var api = new API(settings.apiUrlBase);
-    var origin = utils.getSelfOrigin();
-    if (!origin) {
-      return onResult(errors.InvalidApp('an origin is needed to get products'),
-                      allProducts);
-    }
-    origin = encodeURIComponent(origin);
-    var url;
-
-    if (settings.fakeProducts) {
-      settings.log.warn('about to fetch fake products');
-      url = '/payments/stub-in-app-products/';
-    } else {
-      settings.log.info('about to fetch real products for app',
-                        origin);
-      url = '/payments/' + origin + '/in-app/?active=1';
-    }
-
-    api.get(url, function(err, result) {
-      if (err) {
-        return onResult(err, allProducts);
+  exports.all = function products_all(callback) {
+    var promise = new Promise(function(resolve, reject) {
+      var allProducts = [];
+
+      var api = new API(settings.apiUrlBase);
+      var origin = utils.getSelfOrigin();
+      if (!origin) {
+        return reject(errors.InvalidApp('an origin is needed to get products'));
       }
-      settings.log.info('total products fetched:', result.objects.length);
-      for (var i=0; i < result.objects.length; i++) {
-        var ob = result.objects[i];
-        var productInfo = expandInfo(ob);
-        allProducts.push(productInfo);
+      origin = encodeURIComponent(origin);
+      var url;
+
+      if (settings.fakeProducts) {
+        settings.log.warn('about to fetch fake products');
```

Comment:
Thinking it might be nice to do `var log = settings.log` at the top of this module, and then just use `log.info()` etc.